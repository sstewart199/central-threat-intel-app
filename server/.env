XFORCE_API_KEY=f3dac6ed-5e0d-4be0-9468-881e61647fcf
XFORCE_API_PASS=f1eed86a-2c95-455d-8a50-ccbbe77018f2
ABUSEIP_KEY=60fecf08d7413c7b9118d90816cb5342350ec559ba29d69f03b11fcdf1de341b5f4d561e3d89f155
PORT=3600