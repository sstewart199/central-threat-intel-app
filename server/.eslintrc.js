
module.exports = {
  parser: 'babel-eslint',
  extends: [
    'airbnb/base',
    'plugin:import/recommended',
    'plugin:security/recommended',
    'prettier',
  ],
  parserOptions: {
    ecmaVersion: 6,
    sourceType: 'module',
    ecmaFeatures: {
      experimentalObjectRestSpread: true,
    },
  },
  env: {
    browser: true,
    jest: true,
    node: true,
    es6: true,
  },
  rules: {
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/no-webpack-loader-syntax': 0,
    'no-use-before-define': 0,
    'no-nested-ternary': 0,
    'security/detect-object-injection': 'warn',
    'valid-jsdoc': 2,
    'prettier/prettier': [
      'error',
      {
        singleQuote: true,
        trailingComma: 'es5',
        printWidth: 100,
      },
    ],
  },
  plugins: ['import', 'prettier', 'security'],
  settings: {
    'import/resolver': {
      node: {
        paths: ['src'],
      }
    },
  },
};
