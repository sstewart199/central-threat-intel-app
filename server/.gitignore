# packages
node_modules

# build
build

#dotenv
.env
