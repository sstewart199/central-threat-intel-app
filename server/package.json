{
  "name": "@threat-intel-investigation-app/ui",
  "version": "1.0.0",
  "main": "index.js",
  "author": "Shane Stewart",
  "license": "MIT",
  "scripts": {
    "build": "babel src --out-dir build",
    "dev": "NODE_ENV=development babel-watch src",
    "lint:js:fix": "npm run lint:js --fix",
    "lint:js": "eslint . --ignore-path .gitignore",
    "start": "node build/index.js",
    "test": "jest ./test --no-cache --verbose --env=node",
    "test:coverage": "jest ./test --config ./jest.config.js --coverage --env=node"
  },
  "dependencies": {
    "apollo-server-express": "^2.5.0",
    "axios": "^0.18.0",
    "dotenv": "^8.0.0",
    "express": "^4.16.4",
    "graphql": "^14.3.0",
    "lodash": "^4.17.11",
    "node-fetch": "^2.5.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "apollo-server-testing": "^2.5.0",
    "babel-eslint": "^10.0.1",
    "babel-plugin-import-graphql": "^2.7.0",
    "babel-watch": "^7.0.0",
    "eslint": "5.3.0",
    "eslint-config-airbnb": "17.1.0",
    "eslint-config-prettier": "^4.3.0",
    "eslint-plugin-import": "^2.14.0",
    "eslint-plugin-prettier": "^3.1.0",
    "eslint-plugin-security": "^1.4.0",
    "glob": "^7.1.4",
    "graphql-tag": "^2.10.0",
    "jest": "^24.8.0",
    "jest-specific-snapshot": "^2.0.0",
    "prettier": "^1.17.1"
  }
}
