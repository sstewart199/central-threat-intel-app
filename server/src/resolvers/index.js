import abuseIPResolvers from './abuseIP';
import xforceResolvers from './xforce';

const resolvers = [...abuseIPResolvers, ...xforceResolvers];

export default resolvers;
