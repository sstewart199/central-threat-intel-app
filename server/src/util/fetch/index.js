import { ApolloError } from 'apollo-server-express';
import axios from 'axios';

require('dotenv').config();

const qfetch = (api, opts = {}) => {
  return axios({
    method: opts.method || 'GET',
    url: api,
    params: opts.params,
    headers: opts.headers,
  })
    .then(response => {
      return response;
    })
    .catch(error => {
      /* istanbul ignore next */ throw new ApolloError('API Error', {
        error: error.response.data,
      });
    });
};

export default qfetch;
