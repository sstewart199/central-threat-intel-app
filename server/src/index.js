/* istanbul ignore file */
import { app, server } from './server';

const port = process.env.PORT || 3600;

app.listen({ port }, () => {
  // eslint-disable-next-line no-console
  console.log(`GraphQL server ready at http://localhost:${port}${server.graphqlPath}`);
});
