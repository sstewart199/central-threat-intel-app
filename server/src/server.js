import express from 'express';
import { ApolloServer, makeExecutableSchema } from 'apollo-server-express';
// Import Resolvers, Context , TypeDefs and MockResolvers
import resolvers from './resolvers';
import typeDefs from './typeDefs';

const schema = makeExecutableSchema({
  typeDefs,
  resolvers,
});

const server = new ApolloServer({
  schema,
  mockEntireSchema: false,
  debug: true,
  tracing: true,
  formatError: error => {
    return error;
  },
});

const app = express();

server.applyMiddleware({ app });

export { server, app };
