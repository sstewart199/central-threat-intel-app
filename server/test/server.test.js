import fetch from 'node-fetch';
import { app } from '../src/server';

describe('Express Server', () => {
  let expressApp;
  const port = 3002; // Make sure it doesnt conflict with already running app
  beforeAll(async () => {
    expressApp = app.listen(port);
  });
  it('should serve the GraphQL playground', async () => {
    const res = await fetch(`http://localhost:${port}/graphql?query={__type(name:"Query"){name}}`);
    expect((await res.json()).data).toMatchSnapshot();
  });

  afterAll(() => {
    expressApp.close();
  });
});
