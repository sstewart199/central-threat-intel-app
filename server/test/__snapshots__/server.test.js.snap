// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Express Server should serve the GraphQL playground 1`] = `
Object {
  "__type": Object {
    "name": "Query",
  },
}
`;
